### PR TITLE
[CustomCom] Better display for [p]cc list

### LINKED ISSUE #2104: [V3] Escape markdown properly on Custom Command list
# Command bugs

Yes I found bug

#### Command name

`cc list`

#### What cog is this command from?

customcom

#### What were you expecting to happen?

It list the cc beautifully 

#### What actually happened?

image

#### How can we reproduce this issue?

Just have several custom commands with code block / inline code block markdown

### PR DESCRIPTION
**Pull request by Tobotimus**
Uses a menu, optionally embedded with respect to the embed settings, for scrolling through the custom command list, each cc with a ~50 character preview. Format is purposefully similar to the help menu.

Resolves #2104.

### CODE REVIEW

File: redbot/cogs/customcom/customcom.py [hunk 1]
```diff
@@ -338,24 +339,42 @@ def __init__(self, bot):
...
-
-        if len(_commands) < 1500:
-            await ctx.send(box(_commands))
+            # Don't put a line-break in preview
+            newline_pos = result.find("\n")
+            if newline_pos != -1:
+                result = result[:newline_pos] + "..."
+            # Cut preview to 52 characters max
+            if len(result) > 52:
+                result = result[:49] + "..."
```
Comment (mikeshardmind) [inline redbot/cogs/customcom/customcom.py:357]: I don't like this pagification method with embed. I think this is a good case for field based pagination to allow new lines in previews. While this method is certainly cleaner looking than the previous one, It's also less functional for many purposes.

As an example, I have many CCs in a codiing discord utilizing code blocks. The only thing I get is the language specifier and escaped backticks for those.

**Review by mikeshardmind — CHANGES_REQUESTED** [review]
I don't think these changes are of the quality we should push for. It may make the list cleaner, but significantly less functional. If we need an RC3 to accomplish this, so be it in my mind.

**Comment by Twentysix26** [comment]
I like it, certainly leagues better than the previous form.  
Maybe the issue with the code blocks could be resolved by replacing the new lines characters in the previews with a space.
Minor nitpick and maybe up to personal preference: I think it may look better displayed on two lines for each command
```
cmd1
preview
cmd2
preview
(...)
```

**Review by mikeshardmind — APPROVED** [review]
After discussion in discord, I'm modifying my review to approval.

If nobody else does, I'll eventually be looking to add an additional way to browse CCs myself later with more detail, but this shouldn't block unbreaking the current behavior for.

**Comment by Tobotimus** [comment]
@Twentysix26 I actually hadn't thought of replacing newlines with spaces, I will actually give that a go. For displaying the preview on the next line, I think I personally prefer the current method for similarity with the help menu, but I will do it and post some screenshots FTR.

**Comment by Tobotimus** [comment]
### Screenshots
<h4>
<details>
  <summary>Original PR style embed (click me)</summary>
  
</details>
</h4>

<h4>
<details>
  <summary>With newlines replaced with spaces</summary>
  
</details>
</h4>

<h4>
<details>
  <summary>With preview on following line</summary>
  
</details>
</h4>

I prefer the 2nd one, so I'm gonna go with it :smiley:

**Pull request merged**